MonIOWebApp/ClientApp/src/app/budget/budget-data.service.ts

```diff
@@ -1,30 +1,120 @@
 import { Injectable } from '@angular/core';
 import { MockApiDataService } from 'src/app/mock-services/mock-api-data.service';
 import { BehaviorSubject } from 'rxjs/internal/BehaviorSubject';
-import { Budget} from 'src/app/models/Budget';
+import { Budget, BUDGETEXPENSEENUMS} from 'src/app/models/Budget';
 import { Observable } from 'rxjs/internal/Observable';
+import { BudgetExpense } from 'src/app/models/BudgetExpense';
+import { modelGroupProvider } from '@angular/forms/src/directives/ng_model_group';
 @Injectable({
   providedIn: 'root'
 })
 export class BudgetDataService {
   budgetData: Budget;
   _budgetData: BehaviorSubject<Budget>;
   constructor(private apiData : MockApiDataService) { 
+    this._budgetData = new BehaviorSubject<Budget>(new Budget);
     this.apiData.budgetViewData.subscribe(data => {
       console.log(data);
       if(data !== null || data !== undefined){
         this.budgetData = data;
-        this._budgetData = new BehaviorSubject<Budget>(data);
+        this._budgetData.next(data);
       }else{
         this.budgetData = new Budget;
-        this._budgetData = new BehaviorSubject<Budget>(new Budget);
+        this._budgetData.next(null);
       }
-      
     });
   }
 
   getAllBudgetData():Observable<Budget>{
     return this._budgetData.asObservable();
   }
 
+
+  modifyAnExpense(expense:BudgetExpense, expenseType:string, modificationType:string){
```

Comment:
Shouldn't BudgetExpense have the type as part of the object?

Reply:
The Original BudgetExpenseDTO only has Name, ID, and Value. Can we guarantee that We'll always get the BudgetExpenseDetailedDTO for the initial budget call? If so I can model BudgetExpense.ts off of BudgetExpenseDetailedDTO.cs and we can just use that for everything.